test/common/config/filesystem_subscription_impl_test.cc

```diff
@@ -79,6 +81,222 @@ TEST_F(FilesystemSubscriptionImplTest, UpdateTimeChangedOnUpdateSuccess) {
   EXPECT_TRUE(statsAre(3, 2, 0, 0, 0, TEST_TIME_MILLIS + 1, 7148434200721666028, "0"));
 }
 
+// TODO(htuch): Add generic test harness support for collection subscriptions so that we can test
+// gRPC/HTTP transports similar to below.
+class FilesystemCollectionSubscriptionImplTest : public testing::Test,
```

Comment:
I'm wondering, per my comment above in the prof code about shared code, could the tests here actually be a TEST_P that somehow creates a collection or not and just runs all the same tests for the most part to verify the same stats, etc.?

Reply:
I plan on a followup PR to address this as per the TODO.

Reply:
OK, seems like a lot of duplication but up to you.

Reply:
The more time I spent looking at these tests and what coverage they provide, the more I realized they don't actually do a whole lot for some of the unhappy path cases that I've been adding in this PR. I definitely want to reform this entire set of tests before moving forward with any further UDPA PRs, since they are key to being able to validate implementations for both filesystem and gRPC. This is probably going to be a series of PRs that don't have a ton to do with the filesystem list collection support per se, but they would benefit from having it landed as a forcing function.

Reply:
Sure sounds fine to do as followup cleanups if you are tracking.